lib/spack/spack/package_base.py

```diff
@@ -1669,7 +1669,11 @@ def do_stage(self, mirror_only=False):
         self.stage.create()
 
         # Fetch/expand any associated code.
-        if self.has_code and not self.spec.external:
+        dev_path_var = self.spec.variants.get("dev_path", None)
```

Comment:
I don't think this test is sufficient but I'm not positive. `dev_path` gets populated after concretization. Running `spack develop -p` actually writes a custom path into config, so I think that would be the more appropriate check.

Something like 

```
user_dev_path = spack.config.get(f"develop:{self.name}:path", None)
if self.has_code and not self.spec.external and not user_dev_path:
   self.do_fetch(mirror_only)
```